File: Striver Important Interview Questions/Day 15 - String/roman-to-integer.cpp

// https://leetcode.com/problems/roman-to-integer/

class Solution {
public:
    int romanToInt(string s) {
        
        int ans = 0;
        int n = s.size();
        if(n == 0) return ans;

        unordered_map<char, int> romanToIntegerMapping = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };

        for(int i = 0; i < n; ){
            if(i+1 < n && romanToIntegerMapping[s[i]] < romanToIntegerMapping[s[i+1]]){
                ans += romanToIntegerMapping[s[i+1]] - romanToIntegerMapping[s[i]];
                i+=2;  
            }else{
                ans += romanToIntegerMapping[s[i]];
                i+=1;
            } 
        }

        return ans;
    }
};

File: Striver A2Z/Step 5 - String [Basic and Medium]/Lec 2 - Medium String Problems/roman-to-integer.js

// https://leetcode.com/problems/roman-to-integer/

/**
 * @param {string} s
 * @return {number}
 */

const mapping = {
    'I' : 1,
    'V' : 5,
    'X' : 10,
    'L' : 50,
    'C' : 100,
    'D' : 500,
    'M' : 1000 
}

var romanToInt = function(str) {
    let ans = 0;
    for(let i = 0; i < str.length; ){
        if(mapping[str[i]] < mapping[str[i+1]]){
            ans += (mapping[str[i+1]] - mapping[str[i]]);
            i+=2;
        }else {
            ans += mapping[str[i]];
            i+=1;
        }
    }
    return ans;
};